---
title: '"LDAP超时错误 | AEM Oak”'
description: 描述
solution: Experience Manager
product: Experience Manager
applies-to: "Experience Manager"
keywords: "KCS"
resolution: Resolution
internal-notes: null
bug: false
article-created-by: Roxann McGlumphy
article-created-date: "10/21/2021 9:51:18 PM"
article-published-by: Roxann McGlumphy
article-published-date: "10/21/2021 9:52:58 PM"
version-number: 1
article-number: KA-17498
dynamics-url: "https://adobe-ent.crm.dynamics.com/main.aspx?forceUCI=1&pagetype=entityrecord&etn=knowledgearticle&id=72595e04-b932-ec11-b6e5-000d3a5ba97a"
exl-id: c0406e46-0830-4e6a-b321-0d176e7665ad
source-git-commit: 0c3e421beca46d9fe1952b1f98538a50697216a0
workflow-type: tm+mt
source-wordcount: '432'
ht-degree: 0%

---

# LDAP超时错误 | AEM Oak

## 描述


您已配置 [通过AEM进行LDAP身份验证](https://docs.adobe.com/docs/en/aem/6-2/administer/security/ldap-config.html) 1，且无法允许LDAP用户登录。  您会看到以下日志消息：
<br><br><br><br><br> <br><br> <br><br><br><br>

| `Caused by: org.apache.directory.api.ldap.model.exception.LdapException: TimeOut occurred` `at org.apache.directory.ldap.client.api.LdapNetworkConnection.writeRequest(LdapNetworkConnection.java:4106)` `at org.apache.directory.ldap.client.api.LdapNetworkConnection.bindAsync(LdapNetworkConnection.java:1290)` `at org.apache.directory.ldap.client.api.LdapNetworkConnection.bind(LdapNetworkConnection.java:1188)` `at org.apache.directory.ldap.client.api.AbstractLdapConnection.bind(AbstractLdapConnection.java:127)` `at org.apache.directory.ldap.client.api.AbstractLdapConnection.bind(AbstractLdapConnection.java:112)` `at org.apache.directory.ldap.client.api.DefaultLdapConnectionFactory.bindConnection(DefaultLdapConnectionFactory.java:64)` `at org.apache.directory.ldap.client.api.DefaultLdapConnectionFactory.newLdapConnection(DefaultLdapConnectionFactory.java:107)` `at org.apache.directory.ldap.client.api.ValidatingPoolableLdapConnectionFactory.makeObject(ValidatingPoolableLdapConnectionFactory.java:133)` `at org.apache.directory.ldap.client.api.ValidatingPoolableLdapConnectionFactory.makeObject(ValidatingPoolableLdapConnectionFactory.java:59)` `at org.apache.commons.pool.impl.GenericObjectPool.borrowObject(GenericObjectPool.java:1188)` `at org.apache.directory.ldap.client.api.LdapConnectionPool.getConnection(LdapConnectionPool.java:123)` `at org.apache.jackrabbit.oak.security.authentication.ldap.impl.LdapIdentityProvider.connect(LdapIdentityProvider.java:771)` `... 57 common frames omitted` |
| --- |

<br><br><br><br><br> <br><br>
1 [https://docs.adobe.com/docs/en/aem/6-2/administer/security/ldap-config.html](https://docs.adobe.com/docs/en/aem/6-2/administer/security/ldap-config.html)
<br><br><br><br>原因<br><br><br><br><br><br>
超时错误通常表示LDAP服务器由于防火墙、不可访问（网络问题）、关闭或无响应而无法访问AEM。


## 分辨率


要修复此问题，您需要调试从AEM到LDAP服务器的连接。  以下是一些要检查的内容：

- 检查LDAP服务器是否可从AEM服务器以外的其他计算机使用LDAP浏览器(例如 [[!DNL JXplorer]](http://jxplorer.org/).  如果无法访问，则可能已关闭，或者可能存在网络或防火墙问题。 请与您的网络运营团队和管理LDAP服务器的团队联系以进行调查。
- 如果可以从其他计算机访问LDAP服务器，则从AEM服务器操作系统中进行测试。  在AEM服务器操作系统上安装LDAP客户端，然后尝试从此处访问LDAP服务器。  开 [!DNL Linux]，则可以使用 [ldapsearch](https://access.redhat.com/documentation/en-US/Red_Hat_Directory_Server/8.2/html/Administration_Guide/Examples-of-common-ldapsearches.html) 命令。  开 [!DNL Windows]，使用 [[!DNL JXplorer]](http://jxplorer.org/).
- 如果服务器可以访问LDAP服务器，但基于AEM LDAP的登录失败，则我们需要检查“LDAP身份提供程序”配置。  登录到OSGi Web控制台(http://)*aem-host:port*/system/console/configMgr)并搜索“*[!DNL Apache][!DNL Jackrabbit Oak] LDAP标识提供程序*&quot;  您可以尝试的一些操作可能会解决此问题：
   - 微调“*用户基DN*&quot;, &quot;*用户额外过滤器*&quot;, &quot;*组基本DN*&quot;和&quot;*组附加过滤器*“ ”，以使搜索过滤器仅将相关用户和组返回到AEM。
   - 确保“*绑定DN*&quot;和&quot;*绑定密码*&quot;正确
   - 取消选中“*验证时的管理池查找*&quot;和&quot;*验证时的用户池查找。*&quot;
   - 将“*搜索超时*&quot;


LDAP标识提供程序配置的屏幕截图：
![rtaimage_3_](https://helpx.adobe.com/content/dam/help/en/experience-manager/kb/LDAP-error/jcr%3acontent/main-pars/image/rtaimage_3_.png "rtaimage_3_")
- 对于大多数企业客户，LDAP通常是负载平衡的。 如果位于LDAP服务器前面的负载平衡器由于某些原因已将您的AEM Server IP列入黑名单，则您也可能会面临此问题。 如果出现此问题，请与LDAP团队联系以解决此问题。 作为快速测试，您可能希望直接绕过LDAP负载平衡器点击LDAP服务器IP，以查看AEM中的LDAP身份验证是否成功。
